src/main/java/htsjdk/samtools/cram/compression/range/ByteModel.java

```diff
@@ -0,0 +1,117 @@
+package htsjdk.samtools.cram.compression.range;
+
+import java.nio.ByteBuffer;
+
+public class ByteModel {
+    // spec: To encode any symbol the entropy encoder needs to know
+    // the frequency of the symbol to encode,
+    // the cumulative frequencies of all symbols prior to this symbol,
+    // and the total of all frequencies.
+    public int totalFrequency;
+    public int maxSymbol;
+    public final int[] symbols;
+    public final int[] frequencies;
+
+    public ByteModel(final int numSymbols) {
+        // Spec: ModelCreate method
+        this.totalFrequency = numSymbols;
+        this.maxSymbol = numSymbols - 1;
+        frequencies = new int[maxSymbol+1];
+        symbols = new int[maxSymbol+1];
+        for (int i = 0; i <= maxSymbol; i++) {
+            this.symbols[i] = i;
+            this.frequencies[i] = 1;
+        }
+    }
+
+    // TODO: use this method to reset
+    public void reset() {
+        totalFrequency = 0;
+        for (int i = 0; i <= maxSymbol; i++) {
+            symbols[i] = 0;
+            frequencies[i] = 0;
+        }
+        // maxSymbol = 0; // TODO: ???
```

Comment:
I'd say yes, since you're resetting everything else, but this method appears to be unused. Which is probably good, since the initialization seems to be context dependent anyway given that it requires knowing the number of symbols. So we should either finish it if we need to call it from somewhere, or else remove it.